src/callbacks_stage/subcell_limiter_idp_correction_2d.jl

```diff
@@ -0,0 +1,44 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+@inline function perform_idp_correction!(u, dt, mesh::TreeMesh2D, equations, dg, cache)
```

Comment:
Does inlining change the performance notably? If not, we should probably avoid enforcing it

Reply:
No, it does not. I'll remove it. Thanks.

Reply:
While we are at it, when does it make sense to enforce it with `@inline`?

Reply:
Basically when it improves performance, e.g., for numerical fluxes and other node-level functions. Something that involves threading itself typically does not benefit (since threading creates closures and is already at the level of "loop over all elements/interfaces/.."").